File: programming/5.1/main.cpp
#include <iostream>
#include <cmath>
#include <ctime>
#define n 10
#define k 10


using namespace std;

int randint(int min, int max) {
  return rand() % (max + 1 - min) + min;
}

float generateDataItem() {
  return n * k * randint(-16 * n * randint(0, 10), 10);
}

void fillData(float **data) {
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < k; j++) {
      data[i][j] = generateDataItem();
    }
  }
}

int main() {
  srand(time(NULL));
  float **data = new float*[n];

  for (int i = 0; i < n; i++) {
    data[i] = new float[k];
  }

  fillData(data);

  cout << "Array: " << endl;
  for (int i = 0; i < n; i++) {
    cout << "| ";
    for (int j = 0; j < k; j++) {
      cout << data[i][j] << " ";
    }
    cout << endl;
  }

  // task 16
  int 
    minIndexN = 0,
    minIndexK = 0,
    minValue = data[0][0],

    maxIndexN = 0,
    maxIndexK = 0,
    maxValue = data[0][0];

  for (int i = 0; i < n; i++) {
    for (int j = 0; j < k; j++) {
      if (minValue > data[i][j]) {
        minValue = data[i][j];
        minIndexN = i;
        minIndexK = j;
      }

      if (maxValue < data[i][j]) {
        maxValue = data[i][j];
        maxIndexN = i;
        maxIndexK = j;
      }
    }
  }

  cout << endl;
  cout << "Min index\n  n: " << minIndexN << "\n  k: " << minIndexK << endl;
  cout << "Max index\n  n: " << maxIndexN << "\n  k: " << maxIndexK << endl;
  

  return 0;
}

File: programming/5.1/main.py
from random import randint
from prettytable import PrettyTable

table = PrettyTable()
table.field_names = [f'{i}' for i in range(10)]
data = [[None] * 10] * 10

def generate_data_item():
  return randint(-16 * 10 * randint(0, 10), 10 * 10 * 10)

def generate_data(data):
  for i in range(len(data)):
      data[i] = [generate_data_item() for i in range(len(data[i]))]

  return data


data = generate_data(data)

for i in data:
  table.add_row(i)

min_index_n = 0
min_index_k = 0
min_value = data[0][0]

max_index_n = 0
max_index_k = 0
max_value = data[0][0]


for i in range(len(data)):
  for j in range(len(data[i])):
    if data[i][j] < min_value:
      min_value = data[i][j]
      min_index_n = i
      min_index_k = j

    if data[i][j] > max_value:
      max_value = data[i][j]
      max_index_n = i
      max_index_k = j

print(table)
print()
print(f'Min\n  n:{min_index_n}\n  k:{min_index_k}\n')
print(f'Max\n  n:{max_index_n}\n  k:{max_index_k}')
